### DISCUSSION TITLE
1.27 version - How do I decrypt the access token value in the token table?

### DISCUSSION THREAD

**Discussion by kkangsla** [teslamate-org/teslamate#2851] [Q&A]
Hi.

teslamate 1.27 version

How do I decrypt the access token value in the token table?

I'm use PHP

Thank you very much.

**Comment by dayanfcosta** [comment]
I also would like to use the access and refresh token from teslamate to perform another actions integrating with their database.
I'm using java, I found out about Cipher but I don't know the parameters like algorithm, transformation, etc. 

Could anyone help with this?

**Comment by LelandSindt** [comment]
This is how we did it in https://github.com/tobiasehlert/teslamateapi 

https://github.com/tobiasehlert/teslamateapi/commit/3f9e8c471bf2477ad7827cc208b9f492f4b0bc38
